R/angles.R

```diff
@@ -0,0 +1,101 @@
+calculate.angles.radial <- function(v, tree, spread, total.angle) {
+    root.node.id <- v$id[[1]];
+    node.ids <- c(root.node.id);
+
+    total.angle <- total.angle * spread;
+    angles <- numeric(nrow(tree));
+
+    while (length(node.ids) > 0) {
+        current.node.id <- node.ids[1];
+        node.ids <- node.ids[-1];
+
+        parent.id <- tree$parent[tree$tip == current.node.id];
+
+        if (parent.id == -1) {
+            tree$angle[tree$tip == current.node.id] <- 0;
+            }
+
+        child.ids <- tree$tip[tree$parent == current.node.id & !is.na(tree$parent)];
+        num.children <- length(child.ids);
+
+        if (length(child.ids) > 0) {
+            parent.angle <- parent.angle <- tree$angle[tree$tip == current.node.id];
+            child.weight <- assign.weight(current.node.id, v);
+
+            start.angle <- parent.angle - (total.angle) * (num.children > 1) / 2;
+            num.slices <- max(num.children - 1, 1);
+            angle.increment <- total.angle / num.slices;
+
+            for (i in seq_along(child.ids)) {
+                child.id <- child.ids[i];
+                angle <- start.angle + (i - 1) * (angle.increment);
+                angles[tree$tip == child.id] <- angle;
+                }
+
+            node.ids <- append(node.ids, child.ids);
+            }
+        }
+    
+    angles <- override.angles(tree, v, angles);
+    return(angles);
+    }
+
+calculate.angles.fixed <- function(v, tree, fixed.angle) {
```

Comment:
I tried working through the `testthat` examples and investigating the function with `browser()` but overall I don't understand how the code works.

So feel free to skip me and defer to @aholmes @nwiltsie .

Otherwise if you want to chat more about the code in person/Zoom, I'm happy to help!